crates/turbo-tasks-fetch/src/lib.rs

```diff
@@ -39,13 +46,118 @@ pub async fn fetch(url: StringVc, user_agent: OptionStringVc) -> Result<HttpResp
         builder = builder.header("User-Agent", user_agent);
     }
 
-    let response = builder.send().await?;
-    let status = response.status().as_u16();
-    let body = response.bytes().await?.to_vec();
+    let response = builder.send().await;
+    match response {
+        Ok(response) => {
+            let status = response.status().as_u16();
+            let body = response.bytes().await?.to_vec();
 
-    Ok(HttpResponse {
-        status,
-        body: HttpResponseBodyVc::cell(HttpResponseBody(body)),
+            Ok(HttpResponse {
+                status,
+                body: HttpResponseBodyVc::cell(HttpResponseBody(body)),
+            }
+            .into())
+        }
+        Err(err) => {
+            fetch_issue_from_reqwest_error(&err, context)
+                .cell()
+                .as_issue()
+                .emit();
+            Err(err.into())
+        }
+    }
+}
+
+#[turbo_tasks::value(shared)]
+enum FetchIssueKind {
+    Connect,
+    Timeout,
+    Status(u16),
+    Other,
+}
+
+#[turbo_tasks::value(shared)]
+struct FetchIssue {
+    pub context: FileSystemPathVc,
+    pub url: OptionStringVc,
+    pub kind: FetchIssueKindVc,
+    pub detail: StringVc,
+}
+
+fn fetch_issue_from_reqwest_error(
```

Comment:
This should be a static method on `FetchIssue`. `::from_reqwest_error(url, error)`